packages/qwik/src/core/platform/platform.ts

```diff
@@ -65,9 +65,10 @@ export const createPlatform = (): CorePlatform => {
  * @returns Fully qualified URL.
  */
 export const toUrl = (doc: Document, containerEl: QwikElement, url: string | URL): URL => {
-  const baseURI = doc.baseURI;
+  const baseURI = new URL(doc.baseURI);
   const base = new URL(containerEl.getAttribute('q:base') ?? baseURI, baseURI);
-  return new URL(url, base);
+  const pathUrl = (base.pathname + url).replace(/\/+/g, '/');
+  return new URL(pathUrl, base.origin);
```

Comment:
.origin can be `null`, for example when using Capacitor which uses a base of `capacitor://`